File: java01t/src/step05/Test07.java
package step05;

/* 메서드 호출과 로컬 변수의 관계
 * 1. 메서드에 선언하는 변수 : 로컬 변수
 * 2. 파라미터도 로컬 변수의 일종이다.
 * 3. 메서드 호출 시 선언된 변수는 "스택"이라고 부르는 메모리 영역에 준비된다.
 * 4. 메서드가 호출되면 스택 영역에 그 메서드를 위한 변수들이 준비된다.
 * 5. 메서드 호출이 끝나면 스택 영역에 준비된 메모리가 자동 해제된다.
 */
public class Test07 {

  static void m(int a) {
    int b = a * 2;
    System.out.printf("m(): a = %d, b = %d\n", a, b);//다른 메서드의 변수에 접근 불가!
    a++; 
    System.out.printf("m(): a = %d\n", a);
  }

  public static void main(String[] args) {
    int a = 10;
    m(a); // a 변수 자체를 넘기는 것이 아니라, a 변수에 들어 있는 값을 넘긴다.
    System.out.printf("main() : %d\n", a);

    a = 20;
    m(a);
    System.out.printf("main() : %d\n", a);
    
    a = 30;
    m(a);
    System.out.printf("main() : %d\n", a);
    //System.out.println(b); //다른 메서드의 변수에 접근할 수 없다. 오류!
    
  }

}

File: node1t/src/ex1/test07.js
var http = require('http');
var url = require('url');

// 실습 목표: 요청 파라미터 추출하기
// - 요청과 응답에 대해 사용할 수 있는 메서드 
http.createServer(function(req, res) {
	console.log("------------------------------");
	//그냥 분석하면 query string이 문자열로 추출된다.
	//- 쿼리 스트링을 파라미터 별로 추출하고 두 번째 파라미터를 true로 설정하라!
	var urlInfo = url.parse(req.url, true);
	
	console.log("웹 애플리케이션 경로 = ", urlInfo.pathname);
	
	// 테스트 URL: http://localhost:1337/calc/compute.do?v1=20&v2=30&op=plus
	// 파라미터 값 출력하기
	console.log("v1 = ", urlInfo.query.v1);
	console.log("v2 = ", urlInfo.query.v2);
	console.log("op = ", urlInfo.query.op);
	
    res.writeHead(200, {
    		'Content-Type': 'text/html;charset=UTF-8',
    	});
    res.write('<html><head><title>test07</title></head>\n');
    res.write('<body><h1>test07</h1>\n');
    res.write('<p>안녕하세요!</p>\n');
    res.end('</body></html>\n');
    
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');










